### PR TITLE
T22: Swap Indices 

### PR DESCRIPTION
**Pull request by Nandhini-Sakthi**

### CODE REVIEW

**Comment by tmmvigneshm** [comment]
Add comments and headers as discussed over call. (Ask Ashish if you need any assistance)
Convert your code to classic Program.Main method style.
Also add inline comments wherever (comments inside methods for variables & logics) possible.

**Comment by Nandhini-Sakthi** [comment]
Above mentioned corrections were done snr.

File: Program.cs [hunk 1]
```diff
@@ -1,4 +1,45 @@
...
+      static void Main (string[] args) {
+         Random random = new Random ();
+         int[] numbers = new int[8];
+         for (int i = 0; i < numbers.Length; i++)
+            numbers[i] = random.Next (1, 10);
+         Console.WriteLine ("Original Numbers: " + string.Join (" ", numbers));
+         Console.Write ("Enter two indices to swap (e.g., 2 3): ");
+         string input = Console.ReadLine ();
+         string swappedResult = SwappedNumber (input, numbers);
+         Console.WriteLine ("Swapped Numbers: " + swappedResult);
```
Comment (tmmvigneshm) [inline Program.cs]: Proper warning should be given when two given input indices are same. e.g. 2 2

File: Program.cs [hunk 2]
```diff
@@ -1,4 +1,45 @@
...
+using System;
+namespace Training {
+   #region Program ------------------------------------------------------------------------------
+   /// <summary>Swap Indices</summary>
+   internal class Program {
+      #region Methods ---------------------------------------------
+      /// <summary>Display random series of numbers to the user </summary>
+      /// <param name="args">arguments</param>
+      static void Main (string[] args) {
+         Random random = new Random ();
```
Comment (tmmvigneshm) [inline Program.cs]: simplify by Random random = new ();

File: Program.cs [hunk 3]
```diff
@@ -1,4 +1,45 @@
...
+         Console.Write ("Enter two indices to swap (e.g., 2 3): ");
+         string input = Console.ReadLine ();
+         string swappedResult = SwappedNumber (input, numbers);
+         Console.WriteLine ("Swapped Numbers: " + swappedResult);
+      }
+      /// <summary>swapping the numbers using indices</summary>
+      /// <param name="input">Index</param>
+      /// <param name="numbers">Array of Random numbers</param>
+      /// <returns>swapped number</returns>
+      static string SwappedNumber (string input, int[] numbers) {
```
Comment (tmmvigneshm) [inline Program.cs]: Function name should be logical and "SwapNumbers" sounds good.

File: Program.cs [hunk 4]
```diff
@@ -1,4 +1,45 @@
...
+         string[] indices = input.Split ();
+         if (indices.Length == 2 && int.TryParse (indices[0], out int i) && int.TryParse (indices[1], out int j) && i >= 0 && i < numbers.Length && j >= 0 && j < numbers.Length) {
+            int temp = numbers[i];
+            numbers[i] = numbers[j];
+            numbers[j] = temp;
+            return string.Join (" ", numbers);
+         } return "Invalid input or indices.";
+         #endregion
```
Comment (tmmvigneshm) [inline Program.cs]: Closing of this region has been put in wrong place. Please check.

File: Program.cs [hunk 5]
```diff
@@ -1,4 +1,45 @@
...
+      }
+      #endregion
```
Comment (tmmvigneshm) [inline Program.cs:55]: closing region(endregion) at wrong place.

File: Program.cs [hunk 6]
```diff
@@ -1,4 +1,45 @@
...
+         string swappedResult = SwappedNumber (input, numbers);
+         Console.WriteLine ("Swapped Numbers: " + swappedResult);
+      }
+      /// <summary>swapping the numbers using indices</summary>
+      /// <param name="input">Index</param>
+      /// <param name="numbers">Array of Random numbers</param>
+      /// <returns>swapped number</returns>
+      static string SwappedNumber (string input, int[] numbers) {
+         string[] indices = input.Split ();
+         if (indices.Length == 2 && int.TryParse (indices[0], out int i) && int.TryParse (indices[1], out int j) && i >= 0 && i < numbers.Length && j >= 0 && j < numbers.Length) {
```
Comment (tmmvigneshm) [inline Program.cs]: Try not to parse the input inside the logic region(methods).
In the main method itself you should check if it is valid input or not.

File: Program.cs (hunk 3 above)
Comment (tmmvigneshm) [inline Program.cs]: I would like to see two arguments here e.g. int a , int b where a and b are indices.
string input argument does not look appropriate here.

File: Program.cs (hunk 6 above)
Comment (tmmvigneshm) [inline Program.cs]: This logic is fine and some of the checks can be handled at main method itself while you get the input.

**Pull request closed without merging**